lib/api/multipartDelete.js

```diff
@@ -122,14 +142,26 @@ function multipartDelete(authInfo, request, log, callback) {
                         bucket: bucketName,
                         byteLength: partSizeSum,
                     });
+                    return next(null, destBucket, location);
+                });
+        },
+        function ifMultipleBackend(destBucket, location, next) {
+            if (config.backends.data === 'multiple') {
+                return multipleBackendGateway.abortMPU(objectKey,
+                uploadId, location, destBucket, log, err => {
```

Comment:
Could `location` be undefined here, say, if the object was put without a location constraint originally? If the answer is yes, it could be wise to check this and skip as well if no location exists for the object.